src/hooks/useYoutubeApi.js

```diff
@@ -1,31 +1,29 @@
-import {useState, useEffect, useCallback} from 'react';
-import youtube from '../apis/youtube';
-
-const useYoutubeApi = (query) => {
-  const [videos, setVideos] = useState([]);
-
-  const handleSubmit = useCallback(async (query) => {
-    
-    try{
-      const response = await youtube.get('/search', {
-      params: {
-        q: query,
-        type: 'video' 
-      }
-      })
-      setVideos(response.data.items);
-    } catch(e){
-      console.log(e);
-    }
-    
-  }, [])
-
-  useEffect(()=>{
-    handleSubmit(query)
-  }, [query, handleSubmit])
-
-  return [videos];
-  
-}
-
-export default useYoutubeApi;
\ No newline at end of file
+import { useState, useEffect, useCallback } from 'react';
+import youtube from '../apis/youtube';
+
+const useYoutubeApi = (query) => {
+  const [videos, setVideos] = useState([]);
+
+  const handleSubmit = useCallback(async (search) => {
+    try {
+      const response = await youtube.get('/search', {
+        params: {
+          q: search,
+          type: 'video',
+        },
+      });
+      const videoIds = response.data.items.map((item) => item.id.videoId);
+      setVideos(videoIds);
+    } catch (e) {
+      console.log(e);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (query) handleSubmit(query);
+  }, [query, handleSubmit]);
+
+  return [videos];

```

Comment:
Since this is the only thing you return, consider returning just that without the array.